addons/purchase_stock/wizard/product_replenish.py

```diff
@@ -23,13 +23,18 @@ def default_get(self, fields):
                     *self.env['stock.warehouse']._check_company_domain(company),
                 ], limit=1).id
             orderpoint = self.env['stock.warehouse.orderpoint'].search([('product_id', 'in', [product_tmpl_id.product_variant_id.id, product_id.id]), ("warehouse_id", "=", res['warehouse_id'])], limit=1)
-            res['supplier_id'] = False
-            if orderpoint:
+            if orderpoint.supplier_id:
+                res['route_id'] = orderpoint.route_id.id
                 res['supplier_id'] = orderpoint.supplier_id.id
-            elif product_tmpl_id.seller_ids:
-                res['supplier_id'] = product_tmpl_id.seller_ids[0].id
         return res
 
+    @api.onchange('route_id', 'supplier_id')
+    def _onchange_supplier_id(self):
+        if self.show_vendor and not self.supplier_id:
```

Comment:
Here is a use case raising a traceback:

1. Create a storable product using the buy route without any set vendor
2. Inventory > Operation > Procurement > Replenishment
3. Create a new one for your product
4. Go back to your product form and click on replenish
> Traceback: index out of range.

```suggestion
        if self.show_vendor and not self.supplier_id and self.product_tmpl_id.seller_ids:
```